src/edn/mod.rs

```
@@ -712,10 +709,7 @@ impl std::str::FromStr for Edn {
 
     /// Parses a `&str` that contains an Edn into `Result<Edn, EdnError>`
     fn from_str(s: &str) -> Result<Self, Self::Err> {
-        let clean = String::from(s.trim_start());
```

this had a nice reason to exist, it broke when there was whitespaces before the edn

---

I want to remove it so the walker impl can give accurate feedback.
`"                   {"` should throw a nice error at line 1 column 20, not line 1 column 2

---

that makes sense, one thing that has always annoyed me is the fact that this ` edn.chars().enumerate()` doesnt preserve the initial enumeration, but associating the enumerate number to the iterator, means we actually need to consume it in the first pass. And that is slow

---

I actually think the walker impl won't create an owned string at all. I was mostly just trying to clean things up to move to a 1.0 api (both public and private).